File: CoWorkHub/CoWorkHub.Model/DayAvailability.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CoWorkHub.Model
{
    public class DayAvailability
    {
        public DateTime Date { get; set; }
        public bool IsAvailable { get; set; }
        public int Capacity { get; set; }
        public int Reserved { get; set; }
        public int Free { get; set; }
    }
}

File: CoWorkHub/UI/coworkhub_mobile/lib/models/day_availability.dart
import 'package:json_annotation/json_annotation.dart';

part 'day_availability.g.dart';

@JsonSerializable()
class DayAvailability {
  DateTime date;
  bool isAvailable;
  int capacity;
  int reserved;
  int free;

  DayAvailability({
    required this.date,
    required this.isAvailable,
    required this.capacity,
    required this.reserved,
    required this.free,
  });

  factory DayAvailability.fromJson(Map<String, dynamic> json) =>
      _$DayAvailabilityFromJson(json);

  Map<String, dynamic> toJson() => _$DayAvailabilityToJson(this);
}
